packages/alea-frontend/components/McqComponent.tsx

```diff
@@ -0,0 +1,409 @@
+import { useState, useEffect } from 'react';
+import { Box, Typography, Button, CircularProgress } from '@mui/material';
+import ProblemCard from './ProblemCard';
+import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos';
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+
+import {
+  addReferences,
+  ClickedButtons,
+  extractQuestions,
+  fetchAllVersions,
+  fixDistractor,
+  Problem,
+  removeAmbiguity,
+  ResponseData,
+  VersionData,
+} from '@stex-react/api';
+
+const MCQComponent = ({ respData }) => {
+  const [currentProblemIndex, setCurrentProblemIndex] = useState<number>(0);
+  const [selectedOptions, setSelectedOptions] = useState({});
+  const [showAllProblems, setShowAllProblems] = useState<boolean>(true);
+  const [clickedButtons, setClickedButtons] = useState<ClickedButtons>({});
+  const [currentProblems, setCurrentProblems] = useState<Problem[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [versionsData, setVersionsData] = useState<VersionData[]>([]);
+  const [matchingQuestionData, setMatchingQuestionData] = useState<VersionData | null>(null);
+  const [selectedVersion, setSelectedVersion] = useState<number>(1);
+  const [totalVersions, setTotalVersions] = useState<number>(1);
+  const [responseData, setResponseData] = useState<ResponseData | null>(respData);
+
+  function parseJsonString(jsonString: string): any[] {
+    const cleanedString = jsonString.replace(/^[\s\S]*?###\s*|\s*###[\s\S]*$/g, '');
+
+    try {
+      const jsonObject = JSON.parse(cleanedString);
+      return jsonObject;
+    } catch (error) {
+      console.error('throw new error', error);
+      return [];
+    }
+  }
+
+  useEffect(() => {
+    if (respData !== responseData) {
+      setResponseData(respData);
+    }
+  }, [respData]);
+
```

Comment:
do we need this useEffect ?
You can directly update responseData inside the component if respData is passed as a prop